### PR TITLE
Add item to order

### PR DESCRIPTION
**Pull request by kprakobkit**
A few things:
- need to mock mongoose when testing. i have an x'ed test because i'm not sure how to mock the mongoose model constructor
- using fakerjs to generate auto number for now
- added chai-spies  - take a look in new_order_spec

### CODE REVIEW

File: server/action_creators.js [hunk 1]
```diff
@@ -36,14 +37,37 @@ export function loadOrders() {
...
+      }));
+    });
+  };
+}
+
+export function addOrder(items) {
+  return (dispatch, getState) => {
+    Order({
+      id: faker.random.number(), // need auto generated id...
+      status: constants.OPEN,
```
Comment (achiu8) [inline server/action_creators.js]: Is there a way to default this in the schema definition?

File: server/reducer.js [hunk 2]
```diff
@@ -6,20 +6,10 @@ function setState(state, newState) {
   return Object.assign({}, state, newState);
 }
 
-function addOrder(state, order) {
```
Comment (achiu8) [inline server/reducer.js:9]: I'm starting to wonder if this is an anti-pattern to have only one catch-all reducer.. working okay so far but smells a little.. what do you think?

File: src/components/NewOrder.js [hunk 3]
```diff
@@ -1,26 +1,76 @@
 import { Component, DOM as dom } from 'react';
 import { connect } from 'react-redux';
 import * as actions from '../action_creators';
+import _ from 'underscore';
```
Comment (achiu8) [inline src/components/NewOrder.js:4]: Yo you should check out ramda too. More 'purely' functional, automatically curried functions, easier composition, etc..

http://fr.umio.us/why-ramda/

Comment (achiu8) [inline src/components/NewOrder.js:4]: Although we're not doing anything crazy so far.. so maybe not worth it.

File: src/components/NewOrder.js [hunk 4]
```diff
@@ -1,26 +1,76 @@
...
+    const defaultValue = this.props.items[0];
+    const items = this.state.items.concat(this.state.selectedItem || defaultValue);
+
+    this.setState({ items });
+  }
+
+  selectItem(e) {
+    const itemId = e.target.value;
+    const selectedItem = _.find(this.props.items, (item) => item.id === itemId);
+    this.state.selectedItem = selectedItem;
```
Comment (achiu8) [inline src/components/NewOrder.js]: Should we be using `setState` here?

File: src/components/NewOrder.js [hunk 5]
```diff
@@ -1,26 +1,76 @@
 import { Component, DOM as dom } from 'react';
 import { connect } from 'react-redux';
 import * as actions from '../action_creators';
+import _ from 'underscore';
 
 const mapStateToProps = function (state) {
   return {
-    orders: state.orders,
     items: state.items
```
Comment (achiu8) [inline src/components/NewOrder.js]: What do you think of renaming this collection to `masterItems` or something? Might get confusing to distinguish vs. `state.items`.

File: src/components/NewOrder.js [hunk 6]
```diff
@@ -1,26 +1,76 @@
...
     return dom.div(
       null,
       dom.h1(null, 'New Order Page'),
-      dom.button({
-          className: 'btn btn-default',
-          onClick: this.props.addOrder
-        }, 'Submit')
+      dom.select(
+        { id: 'item', onChange: this.selectItem },
+        this.props.items.map((item) => {
```
Comment (achiu8) [inline src/components/NewOrder.js]: I guess doesn't really matter, but does it not complain about setting `key` for iterated children?

File: src/components/NewOrder.js [hunk 7]
```diff
@@ -1,26 +1,76 @@
...
+          className: 'btn btn-default add-item',
+          onClick: this.addOrderItem
+        },
+        'Add Item'
+      ),
+      dom.div(
+        null,
+        dom.ul(
+          null,
+          this.state.items.map((item) => dom.li({ className: 'addedItem' }, item.name))
```
Comment (achiu8) [inline src/components/NewOrder.js]: Minor: can we use hyphens for classes?

File: test/components/New_Order_spec.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import { expect } from 'chai';
+import React from 'react';
+import chai from 'chai';
```
Comment (achiu8) [inline test/components/New_Order_spec.js:3]: Can we just import `spy` up top?

File: test/server/action_creators_spec.js [hunk 9]
```diff
@@ -89,4 +88,46 @@ describe('server action creators', () => {
...
+      }
+    };
+
+    actions.loadItems()(dispatch).then(() => {
+      expect(dispatched).to.deep.equal(expected);
+      done();
+    });
+  });
+
+  xit('addOrder', () => {
```
Comment (achiu8) [inline test/server/action_creators_spec.js:110]: Hmm.. we could create a new `Order` stub as a constructor (vs. object like above) for this test and proxyquire `actions` by test..? Eh not sure...

File: test/components/New_Order_spec.js (hunk 8 above)
Comment (kprakobkit) [inline test/components/New_Order_spec.js:3]: What do you mean? As opposed to importing it in test_helper?

File: src/components/NewOrder.js (hunk 7 above)
Comment (kprakobkit) [inline src/components/NewOrder.js]: yep that was my bad.

File: test/components/New_Order_spec.js (hunk 8 above)
Comment (achiu8) [inline test/components/New_Order_spec.js:3]: Oh I just meant combine with line 1: `import { expect, spy } from 'chai';`

Then don't need to call `spy()` off `chai` either.

File: server/reducer.js (hunk 2 above)
Comment (kprakobkit) [inline server/reducer.js:9]: I think for right now it's okay since. How do you think of splitting it?

**Pull request merged**